libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -29,265 +27,279 @@
+    void init_xy_controller_stopping_point();
 
-    /// get_alt_target - get desired altitude (in cm above home) from loiter or wp controller which should be fed into throttle controller
-    float get_alt_target() const { return _pos_target.z; }
+    // relax_velocity_controller_xy - initialise the position controller to the current position and velocity with decaying acceleration.
+    ///     This function decays the output acceleration by 95% every half second to achieve a smooth transition to zero requested acceleration.
+    void relax_velocity_controller_xy();
 
-    /// get_alt_error - returns altitude error in cm
-    float get_alt_error() const;
+    /// input_vel_accel_xy calculate a jerk limited path from the current position, velocity and acceleration to an input velocity and acceleration.
+    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
+    ///     The kinematic path is constrained by the maximum acceleration and time constant set using the function set_max_speed_accel_xy and time constant.
+    ///     The time constant defines the acceleration error decay in the kinematic path as the system approaches constant acceleration.
+    ///     The time constant also defines the time taken to achieve the maximum acceleration.
+    ///     The function alters the input velocity to be the velocity that the system could reach zero acceleration in the minimum time.
+    void input_vel_accel_xy(Vector3f& vel, const Vector3f& accel);
+
+    /// input_vel_accel_xy calculate a jerk limited path from the current position, velocity and acceleration to an input position velocity and acceleration.
+    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
+    ///     The kinematic path is constrained by the maximum acceleration and time constant set using the function set_max_speed_accel_xy and time constant.
+    ///     The time constant defines the acceleration error decay in the kinematic path as the system approaches constant acceleration.
+    ///     The time constant also defines the time taken to achieve the maximum acceleration.
+    ///     The function alters the input velocity to be the velocity that the system could reach zero acceleration in the minimum time.
+    void input_pos_vel_accel_xy(Vector3f& pos, Vector3f& vel, const Vector3f& accel);
+
+    // is_active_xy - returns true if the xy position controller has been run in the previous 5 loop times
+    bool is_active_xy() const;
 
-    /// get_vel_z_error_ratio - returns the proportion of error relative to the maximum request
-    float get_vel_z_control_ratio() const { return constrain_float(_vel_z_control_ratio, 0.0f, 1.0f); }
+    /// stop_pos_xy_stabilisation
+    void stop_pos_xy_stabilisation();
```

Comment:
maybe stop_pos_xy_stabilisation() is never called?

Reply:
Used in guided and plane pr's